variables.tf

```diff
@@ -49,6 +55,18 @@ variable "instance_name" {
   type = string
 }
 
+variable "insights_config" {
+  type = object({
+    query_insights_enabled  = bool
+    query_string_length     = optional(number)
+    record_application_tags = optional(bool)
+    record_client_address   = optional(bool)
+  })
+  default = {
+    query_insights_enabled = true
+  }
+}
```

Comment:
... and the actual variable stuff only has 1 default.

Reply:
There's no sense in adding default for the optional keys here. This default only get assigned when the variable isn't passed at all. If any keys are specified, I have to use `defaults()` elsewhere to add the missing defaults.

Reply:
Ah, right. no config vs partial config vs full config supplied. Which requires a 2-step approach for the defaults. Would've been nice if it'd merge in those defaults automagically :wink:

Reply:
Can't, for obvious reasons. In this case it seems easy enough to merge them, but if your variable is a list of objects...